components/layout/query.rs

```diff
@@ -1182,3 +1184,215 @@ fn transform_au_rectangle(
+            }
+        }
+    } else {
+        match target_position {
+            Position::Absolute => {
+                // If the box has position: absolute: The containing block is established
+                // by the nearest ancestor box that establishes an absolute positioning containing block
+                containing_block_chain.insert(target_node.opaque().0, true);
+
+                // Find the nearest ancestor that establishes an absolute positioning containing block
+                let mut found_absolute_cb = false;
+                for &ancestor_node in ancestor_nodes {
+                    if let Some(element) = ancestor_node.as_element() {
+                        let style_data = element.style_data();
+                        let style = style_data.styles.primary();
+
+                        // Check if this ancestor establishes a containing block for absolute descendants
+                        // any positioned element (position != static) or elements with
+                        // transform, will-change, etc. establish containing blocks
+                        if style.establishes_containing_block_for_absolute_descendants(
+                            FragmentFlags::empty(),
+                        ) {
+                            found_absolute_cb = true;
+                            containing_block_chain.insert(ancestor_node.opaque().0, true);
+                            break;
+                        }
+                    }
+                }
+
+                // If no ancestor establishes an absolute positioning containing block,
+                // the absolute positioning containing block is the initial containing block
+                if !found_absolute_cb {
+                    // Include all ancestors up to the initial containing block
+                    for &ancestor_node in ancestor_nodes {
+                        containing_block_chain.insert(ancestor_node.opaque().0, true);
+                    }
+                }
+            },
+            Position::Static => {
+                containing_block_chain.insert(target_node.opaque().0, true);
+
+                // Find the nearest ancestor that establishes a containing block for static descendants
+                for &ancestor_node in ancestor_nodes {
+                    if let Some(element) = ancestor_node.as_element() {
+                        let style_data = element.style_data();
+                        let style = style_data.styles.primary();
+                        let position = style.get_box().position;
+                        let overflow_x = style.get_box().overflow_x;
+                        let overflow_y = style.get_box().overflow_y;
+
+                        // An element establishes a containing block for static descendants if:
+                        // - It has position != static
+                        // - Or it has overflow != visible (establishes BFC)
```

Comment:
That's not true.

If I'm not missing anything, any ancestor that generates a box which isn't either a non-replaced inline or an anonymous block should establish a containing block.

A BFC is a completely unrelated concept, and it's not just established by `overflow`.